compat/ioapi.h

```diff
@@ -4,8 +4,12 @@
 
 #include <stdint.h>
 
-#ifndef _ZLIB_H
-//#include "zlib.h"
+#if !defined(_ZLIB_H) && !defined(ZLIB_H) && !defined(ZLIB_H_)
+#  if __has_include(<zlib-ng.h>)
```

Comment:
I'm afraid `__has_include` is not available in MSVC?

Reply:
Perhaps it will only work if included into C++ file?

Reply:
`__has_include` is in VS 2017 15.3: https://learn.microsoft.com/en-us/cpp/overview/visual-cpp-language-conformance?view=msvc-170

Do we care about obsolete versions of Visual Studio which are more than 7 years old?

Reply:
.

Reply:
@nmoinvaz can we merge this PR as it is, and deal with VS separately?
Otherwise, I guess, all what is needed is to add some nearly empty compat.cc file which just contains `include "compat.h"`? But I'm worried it could affect other platforms which don't expect C++ code.